seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java

```diff
@@ -50,57 +49,57 @@ public class RedisParameters implements Serializable {
     private List<String> redisNodes = Collections.emptyList();
     private long expire = RedisConfig.EXPIRE.defaultValue();
 
-    public void buildWithConfig(Config config) {
+    public void buildWithConfig(ReadonlyConfig config) {
         // set host
-        this.host = config.getString(RedisConfig.HOST.key());
+        this.host = config.get(RedisConfig.HOST);
         // set port
-        this.port = config.getInt(RedisConfig.PORT.key());
+        this.port = config.get(RedisConfig.PORT);
         // set auth
-        if (config.hasPath(RedisConfig.AUTH.key())) {
-            this.auth = config.getString(RedisConfig.AUTH.key());
+        if (config.getOptional(RedisConfig.AUTH).isPresent()) {
+            this.auth = config.get(RedisConfig.AUTH);
         }
         // set db_num
-        if (config.hasPath(RedisConfig.DB_NUM.key())) {
-            this.dbNum = config.getInt(RedisConfig.DB_NUM.key());
+        if (config.getOptional(RedisConfig.DB_NUM).isPresent()) {
+            this.dbNum = config.get(RedisConfig.DB_NUM);
         }
         // set user
-        if (config.hasPath(RedisConfig.USER.key())) {
-            this.user = config.getString(RedisConfig.USER.key());
+        if (config.getOptional(RedisConfig.USER).isPresent()) {
+            this.user = config.get(RedisConfig.USER);
         }
         // set mode
-        if (config.hasPath(RedisConfig.MODE.key())) {
+        if (config.getOptional(RedisConfig.MODE).isPresent()) {
             this.mode =
                     RedisConfig.RedisMode.valueOf(
-                            config.getString(RedisConfig.MODE.key()).toUpperCase());
+                            config.get(RedisConfig.MODE).name().toUpperCase());
         } else {
             this.mode = RedisConfig.MODE.defaultValue();
         }
         // set hash key mode
-        if (config.hasPath(RedisConfig.HASH_KEY_PARSE_MODE.key())) {
+        if (config.getOptional(RedisConfig.HASH_KEY_PARSE_MODE).isPresent()) {
             this.hashKeyParseMode =
                     RedisConfig.HashKeyParseMode.valueOf(
-                            config.getString(RedisConfig.HASH_KEY_PARSE_MODE.key()).toUpperCase());
+                            config.get(RedisConfig.HASH_KEY_PARSE_MODE).name().toUpperCase());
```

Comment:
After `config.get(RedisConfig.HASH_KEY_PARSE_MODE)`. The value already are enum of `HashKeyParseMode`. So we don't need use `valueOf` to parse it again.